src/instructlab/sdg/generate_data.py

```diff
@@ -368,97 +429,301 @@ def generate_data(
+        if leaf_node_type == "knowledge":
+            is_knowledge = True
 
         if is_knowledge:
-            # generate mmlubench data for the current leaf node
             generate_eval_task_data(
                 mmlu_bench_pipe,
                 leaf_node_path,
-                samples,
+                samples_ds,
                 output_dir,
                 date_suffix,
             )
 
+
+def postprocess_taxonomy(
+    input_dir: str,
+    output_dir: str,
+    date_suffix: str,
+    pipeline: Optional[str] = "simple",
+    num_procs: Optional[int] = PipelineContext.DEFAULT_DATASET_NUM_PROCS,
+    system_prompt: Optional[str] = _SYS_PROMPT,
+    use_legacy_pretraining_format: Optional[bool] = True,
+):
+    knowledge_pipe, _, _ = _sdg_init(None, pipeline)
+    mixer = _mixer_init(
+        num_procs,
+        output_dir,
+        date_suffix,
+        knowledge_pipe.auxiliary_inst,
+        system_prompt,
+    )
+
+    input_files = glob.glob(f"{input_dir}/*.jsonl")
+    output_dir = Path(output_dir)
+    output_dir.mkdir(exist_ok=True)
+
+    output_file_messages = f"messages_{date_suffix}.jsonl"
+    output_file_train = f"train_{date_suffix}.jsonl"
+
+    all_generated_data = []
+    for input_file in input_files:
+        logger.debug(
+            "Postprocessing generated taxonomy date in input file: %s", input_file
+        )
+        samples = jlload(input_file)
+        if not samples:
+            raise GenerateException(
+                "Error: No samples found in input file {input_file}"
+            )
+        # For now we assume every sample in the file is the same type
+        first_sample = samples[0]
+        leaf_node_path, leaf_node_type = _extract_leaf_node_path_and_type(first_sample)
+        is_knowledge = False
+        if leaf_node_type == "knowledge":
+            is_knowledge = True
+
+        samples_ds = Dataset.from_list(samples)
+        logger.debug("Postprocessing from samples: %s", samples_ds)
+        all_generated_data.append(samples_ds)
+
         mixer.collect(
             leaf_node_path,
-            new_generated_data,
+            samples_ds,
             is_knowledge,
             use_legacy_pretraining_format,
         )
 
     _gen_train_data(
-        generated_data,
+        all_generated_data,
         os.path.join(output_dir, output_file_train),
         os.path.join(output_dir, output_file_messages),
         system_prompt,
     )
 
-    mixer.generate()
+    mixer.write_recipes()
+
+
+def mix_datasets(
+    recipe_file: str,
+    output_file: str,
+    num_proc: Optional[int] = 8,
+):
+    recipe = Recipe(recipe_file)
+    if recipe.datasets:
+        recipe.save_mixed_dataset(output_file, num_proc)
+    else:
+        logger.info("Not mixing empty recipe file: %s", recipe_file)
+
+
+# This is part of the public API, and used by instructlab.
+# TODO - parameter removal needs to be done in sync with a CLI change.
+# to be removed: logger
+def generate_data(
```

Comment:
the only concern from the core side of things is that this contract is the same as it was - which it appears to be. Thanks!